cs-minibot-platform-src/src/main/resources/public/js/view.js

```diff
@@ -355,4 +421,91 @@ window.addEventListener("keydown", function(e) {
     }
 }, false);
 
+function displayOccupancyMatrix(height, width, size) {
+
+    $.ajax({
+        method: "POST",
+        url: '/getOccupancyMatrix',
+        dataType: 'json',
+        data: JSON.stringify({
+            height:height,
+            width: width,
+            size: size}),
+        contentType: 'application/json',
+        success: function(data) {
+
+            occupancyMatrix = data;
+            occupancyMatrix = padOccupancyMatrix();
+
+            $.ajax({
+                method: "POST",
+                url: '/getDijkstras',
+                dataType: 'json',
+                data: JSON.stringify({
+                matrix: occupancyMatrix}),
+                contentType: 'application/json',
+                success: function(data) {
+                    path = data;
+                    fillOccupancyMatrix(scale, xOffset, yOffset);
+                }
+            });
+
+        }
+
+    });
+}
+
+
+function padOccupancyMatrix() {
+
+    var temp = [];
+    for(var i = 0; i < occupancyMatrix.length; i++) {
+        temp.push([]);
+        for(var j = 0; j < occupancyMatrix[0].length; j++) {
+            temp[i].push(occupancyMatrix[i][j]);
+        }
+    }
+    for(var i = 0; i < occupancyMatrix.length; i++) {
+        for(var j = 0; j < occupancyMatrix.length; j++) {
+            if(occupancyMatrix[i][j] === 1) {
+                if(i-1 >= 0) {
+                    temp[i-1][j] = 1;
+                }
+                if(i+1 < occupancyMatrix.length) {
+                    temp[i+1][j] = 1;
+                }
+                if(j-1 >= 0) {
+                    temp[i][j-1] = 1;
+                }
+                if(j+1 < occupancyMatrix[0].length) {
+                    temp[i][j+1] = 1;
+                }
+                if(j-1 >= 0 && i-1 >= 0) {
+                    temp[i-1][j-1] = 1;
+                }
+                if(j+1 < occupancyMatrix[0].length && i+1 < occupancyMatrix.length) {
+                    temp[i+1][j+1] = 1;
+                }
+                if(j+1 < occupancyMatrix[0].length && i-1 > 0) {
+                    temp[i-1][j+1] = 1;
+                }
+                if(j-1 > 0 && i+1 < occupancyMatrix.length) {
+                    temp[i+1][j-1] = 1;
+                }
+            }
+        }
+    }
+    return temp;
+}
+
+$("#showOccupancyMatrix").click( function() {
+        displayOccupancyMatrix(40, 40, 1.0);
+        setTimeout(function(){
+            fillOccupancyMatrix(scale, xOffset, yOffset);
+            displayBots(bots,scale, xOffset, yOffset);
+            grid.render(stage);
+        }, 3000);
```

Comment:
why are you setting a timeout to do this here? 3000 ms seems like you're trying to predict a delay. there is probably a more direct way to do what you're trying to do.